qt/scientific_interfaces/Inelastic/Manipulation/InelasticDataManipulationElwinTab.cpp

```diff
@@ -339,22 +310,23 @@ void InelasticDataManipulationElwinTab::newInputFilesFromDialog(IAddWorkspaceDia
  * @param index Index of the new selected file
  */
 
-void InelasticDataManipulationElwinTab::checkNewPreviewSelected(int index) {
+void InelasticDataManipulationElwinTab::handlePreviewIndexChanged(int index) {
   auto const workspaceName = m_view->getPreviewWorkspaceName(index);
   auto const filename = m_view->getPreviewFilename(index);
 
-  if (!workspaceName.isEmpty()) {
-    if (!filename.isEmpty())
+  if (!workspaceName.empty()) {
+    if (!filename.empty())
       newPreviewFileSelected(workspaceName, filename);
     else
       newPreviewWorkspaceSelected(workspaceName);
   }
 }
 
-void InelasticDataManipulationElwinTab::newPreviewFileSelected(const QString &workspaceName, const QString &filename) {
+void InelasticDataManipulationElwinTab::newPreviewFileSelected(const std::string &workspaceName,
+                                                               const std::string &filename) {
   auto loadHistory = m_view->isLoadHistory();
-  if (loadFile(filename, workspaceName, -1, -1, loadHistory)) {
-    auto const workspace = getADSMatrixWorkspace(workspaceName.toStdString());
+  if (loadFile(QString::fromStdString(filename), QString::fromStdString(workspaceName), -1, -1, loadHistory)) {
```

Comment:
Would it be possible to refactor the `loadFile` function so that it takes `std::string` parameters rather than `QString` parameters?

Reply:
It is possible, but loadFile is called from other interfaces. So I would have to cast QString to std::string in other interfaces. Is that ok?

Reply:
Yes please do!